packages/api/models/granules.js

```diff
@@ -41,6 +51,33 @@ class Granule extends Manager {
     });
   }
 
+  async get(...args) {
+    return translateGranule(await super.get(...args));
+  }
+
+  async batchGet(...args) {
+    const result = cloneDeep(await super.batchGet(...args));
+
+    result.Responses[this.tableName] = await Promise.all(
+      result.Responses[this.tableName].map(translateGranule)
+    );
+
+    return result;
+  }
+
+  async scan(...args) {
+    const scanResponse = await super.scan(...args);
+
+    if (scanResponse.Items) {
+      return {
+        ...scanResponse,
+        Items: await Promise.all(scanResponse.Items.map(translateGranule))
+      };
```

Comment:
Why `cloneDeep` the `batchGet` result but not the `scanResponse`? I assumed it was to protect the original object, which `cloneDeep` does, but the spread operator does not.

Reply:
I chose to use `cloneDeep` because the objects that were being modified were deep enough in the result that using spread operators was unwieldy.  (I did actually implement it using spread operators, then chose to use `cloneDeep` instead for clarity.  Functionally, I liked the solution using spread operators since it resulted in less duplication of data, but decided it wasn't worth the loss of clarity.)

The equivalent solution using spread operators was:

```js
async batchGet(...args) {
  const result = await super.batchGet(...args);

  return {
    ...result,
    Responses: {
      ...result.Responses,
      [this.tableName]: await Promise.all(result.Responses[this.tableName].map(translateGranule))
    }
  };
}
```

Reply:
> Why `cloneDeep` the `batchGet` result but not the `scanResponse`? I assumed it was to protect the original object, which `cloneDeep` does, but the spread operator does not.

The spread operator does protect the original object.  From [Spread in object literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals), "The Rest/Spread Properties for ECMAScript proposal (stage 4) adds spread properties to object literals. It copies own enumerable properties from a provided object onto a new object."

Reply:
I see, I guess I misunderstood `copies`. Nested objects are still copied by reference. So the spread doesn't make _deep_ clones, only shallow ones, but `scanResponse` is shallow enough that you can handle that. Okay 👍